packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts

```diff
@@ -0,0 +1,195 @@
+import { FastifyPluginAsyncTypebox } from '@fastify/type-provider-typebox';
+import { getAiProviderLanguageModel } from '@openops/common';
+import { logger } from '@openops/server-shared';
+import {
+  DeleteChatHistoryRequest,
+  NewMessageRequest,
+  OpenChatMCPRequest,
+  OpenChatResponse,
+  openOpsId,
+  PrincipalType,
+} from '@openops/shared';
+import {
+  CoreAssistantMessage,
+  CoreToolMessage,
+  pipeDataStreamToResponse,
+  streamText,
+  TextPart,
+} from 'ai';
+import { StatusCodes } from 'http-status-codes';
+import { encryptUtils } from '../../helper/encryption';
+import { aiConfigService } from '../config/ai-config.service';
+import {
+  createChatContext,
+  deleteChatHistory,
+  generateChatIdForMCP,
+  getChatContext,
+  getChatHistory,
+  MCPChatContext,
+  saveChatHistory,
+} from './ai-chat.service';
+import { getMcpSystemPrompt, getSystemPrompt } from './prompts.service';
+
+export const aiMCPChatController: FastifyPluginAsyncTypebox = async (app) => {
+  app.post(
+    '/open',
+    OpenChatOptions,
+    async (request, reply): Promise<OpenChatResponse> => {
+      const chatContext: MCPChatContext = {
+        chatId: request.body.chatId ?? openOpsId(),
+      };
```

Comment:
`chatId` is optional for now, the FE will control if it needs a persistent session